hazelcast/src/main/java/com/hazelcast/internal/diagnostics/DiagnosticsLogFile.java

```diff
@@ -107,36 +104,22 @@ private void renderStaticPlugins() throws IOException {
     }
 
     private void renderPlugin(DiagnosticsPlugin plugin) throws IOException {
-        logWriter.clean();
-        plugin.run(logWriter);
-        //bufferedWriter.append()
-        int desiredLength = charBuff.length;
-        int actualSize = logWriter.length();
-        while (desiredLength < actualSize) {
-            desiredLength *= 2;
-        }
+        logWriter.init(printWriter);
 
-        if (desiredLength != charBuff.length) {
-            charBuff = new char[desiredLength];
-        }
-
-        logWriter.copyInto(charBuff);
-        bufferedWriter.write(charBuff, 0, actualSize);
-        fileLength += actualSize;
+        plugin.run(logWriter);
     }
 
-    private BufferedWriter newWriter() throws FileNotFoundException {
+    private PrintWriter newWriter() throws FileNotFoundException {
         FileOutputStream fos = new FileOutputStream(file, true);
         CharsetEncoder encoder = Charset.forName("UTF-8").newEncoder();
-        return new BufferedWriter(new OutputStreamWriter(fos, encoder));
+        return new PrintWriter(new BufferedWriter(new OutputStreamWriter(fos, encoder), Short.MAX_VALUE));
```

Comment:
We solved this, it's using try(...) which I didn't see because I wasn't expecting Java 7+ (but it's in the JDK). So it's fine to just close the outer resource. Thanks for that lesson, will help me in my Near Cache pre-loader PR.